lib/railway_ipc/storage.ex

```diff
@@ -0,0 +1,37 @@
+defmodule RailwayIpc.Storage do
+  @moduledoc """
+  Behaviour specification for message persistence.
+
+  Any message persistence adapters must conform to this behaviour. Railway
+  ships with a Ecto adapter, but you may define your own. For example, you
+  may define your own no-op implementation if you do not wish to persist
+  messages. You can then override the Railway configuration to use your
+  custom adapter.
+
+  ```
+  config :railway_ipc,
+    storage: MyCustomAdapter
+  ```
+
+  Note that this behaviour is incomplete, it only handles outgoing (published)
+  messages. This is part of an ongoing refactoring to clean up the internal
+  API of Railway. Support for incoming (consumed) messages will be added at a
+  later date.
+
+  """
+
+  defmodule OutgoingMessage do
```

Comment:
Why are we defining outgoing message in here instead of pulling the definition from the outgoing message itself? Or is this just because this is 'in the works' still and it will be moved eventually?

Reply:
Not sure what you mean by "pulling the definition from the outgoing message". This `Storage` module acts as a boundary to the persistence logic. `OutgoingMessage` is the data shape that the boundary deals with, so any information that crosses the boundary has to be a `OutgoingMessage`. Once the consumer side is refactored, there will most likely be another definition for an `IncomingMessage`. The purpose of these definitions is so that the callers of any `Storage` functions deal only with `OutgoingMessage` (and eventually `IncomingMessage`) which prevents persistence logic from seeping back into the domain. Does that answer your question?

Reply:
Got it, I think I was getting this confused with Consumed/Published Message classes. Thanks!